src/main/java/org/wildfly/common/lock/SpinLock.java

```diff
@@ -34,24 +37,44 @@
  */
 public class SpinLock implements ExtendedLock {
     private static final long ownerOffset;
+    private static final int defaultSpinLimit;
 
     static {
         try {
             ownerOffset = unsafe.objectFieldOffset(SpinLock.class.getDeclaredField("owner"));
         } catch (NoSuchFieldException e) {
             throw new NoSuchFieldError(e.getMessage());
         }
+        defaultSpinLimit = AccessController.doPrivileged(
+            (PrivilegedAction<Integer>) () -> Integer.valueOf(
+                System.getProperty("jboss.spin-lock.limit", ProcessorInfo.availableProcessors() == 1 ? "0" : "5000")
+            )
+        ).intValue();
```

Comment:
Should the value be validated in case the `jboss.spin-lock.limit` is set to something other than 0 if there is only a single processor?

Reply:
I'm not 100% sure that this would be an error condition TBH.  I have been having a hard time finding applicable research for some reason.

Reply:
I think it should really be fine as you'd have to know to set the system property. The only reason I question it would be someone copying a dev environment to say some smaller instance where there would be only 1 processor. It's probably not all that common though.